sdk/tables/data-tables/src/TablesBatch.ts

```diff
@@ -0,0 +1,321 @@
+  RequestPolicyOptions,
+  WebResource,
+  WebResourceLike,
+  URLBuilder,
+  ServiceClient,
+  RequestPrepareOptions,
+  RawHttpHeaders
+} from "@azure/core-http";
+import {
+  DeleteTableEntityOptions,
+  TableEntity,
+  UpdateMode,
+  UpdateTableEntityOptions
+} from "./models";
+import { TableClient } from "./TableClient";
+import { TablesSharedKeyCredentialLike } from "./TablesSharedKeyCredential";
+import { getAuthorizationHeader } from "./TablesSharedKeyCredentialPolicy";
+import { HeaderConstants } from "./utils/constants";
+
+export interface TablesBatch {
+  partitionKey: string;
+  createEntities: <T extends object>(entitites: TableEntity<T>[]) => Promise<void>;
+  createEntity: <T extends object>(entity: TableEntity<T>) => Promise<void>;
+  deleteEntity: (
+    partitionKey: string,
+    rowKey: string,
+    options?: DeleteTableEntityOptions
+  ) => Promise<void>;
+  updateEntity: <T extends object>(
+    entity: TableEntity<T>,
+    mode: UpdateMode,
+    options?: UpdateTableEntityOptions
+  ) => Promise<void>;
+  submitBatch: () => Promise<TableBatchResponse>;
+}
+
+interface InnerBatchRequest {
+  createPipeline: () => void;
+  appendSubRequestToBody: (request: WebResource) => void;
+  getHttpRequestBody(): string;
+  getMultipartContentType(): string;
+}
+
+export interface TableBatchResponse {
+  responseCount: number;
+  getResponseForEntity: () => HttpResponse;
+}
+
+/**
+ * Creates a new Batch to collect sub-operations that can be submitted together via submitBatch
+ * @param url Tables account url
+ * @param tableName name of the table to target the operations
+ * @param partitionKey partition key
+ * @param credential credential to authenticate the batch request
+ */
+export function createBatch(
+  url: string,
+  tableName: string,
+  partitionKey: string,
+  credential?: TablesSharedKeyCredentialLike
+): TablesBatch {
+  const batchGuid = generateUuid();
+  const batchRequest = createInnerBatchRequest(batchGuid);
+
+  // Creates the pipeline that would intercept the requests
+  const pipeline = batchRequest.createPipeline();
+
+  // Client used to intercept the requests and add them to the batch instead of sending them to the service
+  const interceptClient: TableClient = new TableClient(url, tableName, {
+    requestPolicyFactories: pipeline
+  });
+
+  let batchUrl = url;
+
+  // Depending on the auth method used we need to build the url
+  if (!credential) {
+    // When authenticating with SAS we need to add the SAS token after $batch
+    const urlParts = url.split("?");
+    const baseUrl = urlParts[0];
+    const sas = urlParts.length > 1 ? `?${urlParts[1]}` : "";
+    batchUrl = `${baseUrl}/$batch${sas}`;
+  } else {
+    // When using a SharedKey credential no SAS token is needed
+    batchUrl = `${batchUrl}/$batch`;
+  }
+
+  return {
+    partitionKey,
+    async createEntity<T extends object>(entity: TableEntity<T>): Promise<void> {
+      if (entity.partitionKey !== this.partitionKey) {
```

Comment:
seems like you could make a helper function to check this given an entity. You also don't need to use `this` here, since `partitionKey` is a scope variable.